src/pages/QuestionPage.jsx

```diff
@@ -254,26 +47,36 @@ const QuestionPage = () => {
   };
 
   // Update the question status on answer click (either correct or incorrect)
-  const handleAnswerClickWithStatus = (isCorrect, questionIndex) => {
+  const handleAnswerClickWithStatus = (questionIndex) => {
+    const isCorrect = questionIndex === questionData.correctAnswer;
+    console.log({ isCorrect });
+
     const updatedStatus = [...questionStatus];
-    updatedStatus[questionIndex] = isCorrect ? "correct" : "incorrect";
+    console.log("before", updatedStatus);
+
+    updatedStatus[currentQuestionIndex] = isCorrect ? "correct" : "incorrect";
+    console.log("after", updatedStatus);
     setQuestionStatus(updatedStatus);
 
+    console.log("Updated status after click:", updatedStatus);
+
     // Mark that an answer has been clicked
     setIsAnswerClicked(true);
 
     // Proceed with original answer click handling
-    handleAnswerClick(isCorrect, questionIndex);
+    handleAnswerClick(isCorrect, currentQuestionIndex);
   };
 
-  // Update the status if the time runs out
   const handleQuestionTimeOutWithStatus = useCallback(
     (questionIndex) => {
       if (!isAnswerClicked) {
         const updatedStatus = [...questionStatus];
-        updatedStatus[questionIndex] = "incorrect"; // If time runs out, consider incorrect
+        updatedStatus[questionIndex] = "timeout"; // Mark the status as timeout
         setQuestionStatus(updatedStatus);
-        handleQuestionTimeOut(questionIndex);
+
+        console.log("Updated status after timeout:", updatedStatus);
+
+        handleQuestionTimeOut(questionIndex); // Proceed with original logic
       }
```

Comment:
const handleNextQuestionWithReset = () => {
    setIsAnswerClicked(false); // Reset answer clicked state
    setCurrentQuestionIndex((prevIndex) => prevIndex + 1); // Increment to the next question
    handleNextQuestion(); // Call the existing next question handler
  };
  put this in context